File: HybridMvcSite/src/app/Models/recipeModels.ts
import { Id } from '../Models/Generics/id';

export class RecipeModel {
  id: Id<RecipeModel>;
  name: string;
  description: string;
  ingredients: IngredientModel[];
  instructions: string;
}

export class IngredientModel {
  amount: number;
  instrument: MeasuringTypeModel;
  name: string;
}

export interface DropdownViewModel {
  value: string;
  viewValue: string;
}

export enum MeasuringTypeModel {
  Dash = 1,
  Teaspoon,
  Tablespoon,
  Cup,
  Gallon,
  ToTaste,
  Item
}

File: HybridMvcSite/Models/RecipeModels.cs
using System.Collections.Generic;

namespace HybridMvcSite.Models
{
    public class RecipeModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<IngredientModel> Ingredients { get; set; }
        public string Description { get; set; }
        public string Instructions { get; set; }
    }

    //Any of the foods or substances that are combined to make a particular dish. 
    public class IngredientModel
    {
        public string Name { get; set; }
        public double Amount { get; set; }
        public MeasuringTypeModel Instrument { get; set; }
    }

    public enum MeasuringTypeModel
    {
        Dash,
        Teaspoon,
        Tablespoon,
        Cup,
        Gallon,
        ToTaste,
        Item
    }
}
